mlflow/data/numpy_dataset.py

```diff
@@ -0,0 +1,156 @@
+import hashlib
+import json
+from typing import List, Optional, Any, Dict, Union
+
+import numpy as np
+import pandas as pd
+from functools import cached_property
+
+from mlflow.data.dataset import Dataset
+from mlflow.data.dataset_source import DatasetSource
+from mlflow.data.filesystem_dataset_source import FileSystemDatasetSource
+from mlflow.data.pyfunc_dataset_mixin import PyFuncConvertibleDatasetMixin, PyFuncInputsOutputs
+from mlflow.types import Schema
+from mlflow.types.utils import _infer_schema
+
+
+class NumpyDataset(Dataset, PyFuncConvertibleDatasetMixin):
+    """
+    Represents a NumPy dataset for use with MLflow Tracking.
+    """
+
+    def __init__(
+        self,
+        features: Union[np.ndarray, List[np.ndarray], Dict[str, np.ndarray]],
+        source: FileSystemDatasetSource,
+        targets: Union[np.ndarray, List[np.ndarray], Dict[str, np.ndarray]] = None,
```

Comment:
Can we add a `targets` method to this class that returns the targets array (or None)?